.github/workflows/static.yml

```diff
@@ -38,27 +39,98 @@ jobs:
+        uses: actions/configure-pages@37538b280cbfab35da9dcd936a2b95f1f522d927
+      - name: Get Package version
+        id: get-package-version
+        run: |
+          echo "package_version=$(cat package.json | jq -r '.version')" >> $GITHUB_OUTPUT
+      - name: Get latest release version
+        id: get-latest-release-version
+        uses: actions/github-script@98814c53be79b1d30f795b907e553d8679345975
+        with:
+          github-token: ${{ secrets.GITHUB_TOKEN }}
+          result-encoding: string
+          script: |
+            try {
+              const getReleaseResp = await github.rest.repos.getLatestRelease({
+                owner: context.repo.owner,
+                repo: context.repo.repo
+              })
+              return getReleaseResp.data.name
+            } catch(err) {
+              if (err.name == 'HttpError') {
+                console.warn("Found HttpError")
+                if (err.status == 404)  {
+                  console.error("Error 404: No previous GitHub Releases found. Defaulting to 0.0.0")
+                  return "0.0.0"
+                }
+              } else {
+                console.error(`Failed to get the latest release: ${err.message}`)
+                throw err
+              }
+            }
+      - name: Next release version
+        id: next-version
+        uses: alphagov/pay-ci/actions/next-semver@master
+        with:
+          package_version: ${{ steps.get-package-version.outputs.package_version }}
+          release_version: ${{ steps.get-latest-release-version.outputs.result }}
       - name: Install and Build
         run: |
           npm ci
           bundle exec middleman build
-      - name: Setup Pages
-        uses: actions/configure-pages@37538b280cbfab35da9dcd936a2b95f1f522d927
+          cat <<EOF > ./build/package.json
+          {
+            "name": "pay-product-page",
+            "version": "${{ steps.next-version.outputs.version }}"
+          }
+          EOF
+      - name: Set artifact name
+        id: set-artifact-name
+        run: echo "name=pay-product-page-${{ steps.next-version.outputs.version }}" >> $GITHUB_OUTPUT
       - name: Prepare archive for upload
+        id: archive-assets
         shell: sh
         run: |
           chmod -c -R +rX "build"
           tar \
             --dereference --hard-dereference \
             --directory "build" \
-            -cvf "$RUNNER_TEMP/artifact.tar" \
+            -cvf "${{ steps.set-artifact-name.outputs.name }}.tar" \
             .
       - name: Upload artifact
         uses: actions/upload-artifact@0b7f8abb1508181956e8e162db84b466c27e18ce
         with:
           name: 'github-pages'
-          path: ${{ runner.temp }}/artifact.tar
+          path: ${{ steps.set-artifact-name.outputs.name }}.tar
           retention-days: 7
       - name: Deploy to GitHub Pages
         id: deployment
         uses: actions/deploy-pages@0243b6c10d06cb8e95ed8ee471231877621202c0
+      - name: Create Release
```

Comment:
So we're creating the release *after* we deploy to Github Pages? It feels like this should be the other way around?

Reply:
The deployment will be more important once we've stopped serving `product-pages` on `frontend`. 

At that point, we'll just be tagging the repo rather than releasing a separate asset. This is here just to maintain compatibility in the interim so I've demoted its priority in the workflow

Reply:
I've amended the priority